partiql-spi/src/main/kotlin/org/partiql/spi/connector/sql/SqlConnector.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates.  All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at:
+ *
+ *       http://aws.amazon.com/apache2.0/
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+ *  language governing permissions and limitations under the License.
+ */
+
+package org.partiql.spi.connector.sql
+
+import org.partiql.spi.connector.Connector
+import org.partiql.spi.connector.ConnectorBindings
+import org.partiql.spi.connector.ConnectorFnProvider
+import org.partiql.spi.connector.ConnectorMetadata
+import org.partiql.spi.connector.ConnectorSession
+import org.partiql.spi.connector.sql.info.InfoSchema
+import org.partiql.spi.fn.FnExperimental
+
+/**
+ * An SQL-99 based [Connector] implementation.
+ */
+public abstract class SqlConnector : Connector {
+
+    /**
+     * Default SQL-99 INFORMATION_SCHEMA for use in function resolution.
+     */
+    public open val info: InfoSchema = InfoSchema.default()
+
+    /**
+     * Returns an implementation of [ConnectorMetadata] which provides the INFORMATION_SCHEMA and delegates
+     * to a user-provided [ConnectorMetadata] implementation.
+     *
+     * @param session
+     * @return
+     */
+    abstract override fun getMetadata(session: ConnectorSession): SqlMetadata
+
+    override fun getBindings(): ConnectorBindings = SqlBindings(info)
+
+    @FnExperimental
+    override fun getFunctions(): ConnectorFnProvider = SqlFnProvider(info.functions)
```

Comment:
Per my understanding: 
I believe that the one of the reasons why those class are open is to allow customer to have those built in function available. 

However, with the currently way that the getFunction is implemented, the way to supply a UDF maybe to have your own implementation on ConnectorFunctionProvider. 


```
    @OptIn(FnExperimental::class)
    internal class TestFunctionProvider(val index: FnIndex, val info: FnIndex) : ConnectorFnProvider {
        @FnExperimental
        override fun getFn(path: ConnectorPath, specific: String): Fn? {
            val fn = index.get(path, specific)
            if (fn is Fn) return fn
            val buildin = info.get(path, specific)
            if (buildin is Fn) return buildin
            return null
        }
    }
```

I think it will be better if we provide APIs to combine function index? Or am I missing something here?